modules/experimental-layers/src/tile-layer/utils/tile-cache.js

```diff
@@ -0,0 +1,56 @@
+import Tile from './tile';
+import {getTileIndices} from './viewport-util';
+
+/**
+ * Manages loading and purging of tiles data
+ */
+
+export default class TileCache {
+  constructor({fetchData, size = 10, onUpdate}) {
+    this.fetchData = fetchData;
+    this.size = size;
+    this.onUpdate = onUpdate;
+
+    this.cache = [];
+  }
+
+  finalize() {
+    this.cache = null;
+    this.onUpdate = () => {};
+  }
+
+  update(viewport, callback) {
+    const {cache, size, fetchData} = this;
+    this.onUpdate = callback;
+    const tiles = getTileIndices(viewport).map(({x, y, z}) => {
+      let tile = this._find(x, y, z);
+      if (!tile) {
+        tile = new Tile({
+          fetchData,
+          x,
+          y,
+          z
+        });
+        this._push(tile);
+      }
+      return tile;
+    });
+
+    while (cache.length > size) {
+      cache.shift();
```

Comment:
This is not the correct logic (the fist tile may still be visible).
Add a TODO here if this is not going to be addressed in the current PR.